common/djangoapps/third_party_auth/settings.py

```diff
@@ -1,51 +1,15 @@
 """Settings for the third-party auth module.
 
-Defers configuration of settings so we can inspect the provider registry and
-create settings placeholders for only those values actually needed by a given
-deployment. Required by Django; consequently, this file must not invoke the
-Django armature.
-
 The flow for settings registration is:
 
 The base settings file contains a boolean, ENABLE_THIRD_PARTY_AUTH, indicating
-whether this module is enabled. Ancillary settings files (aws.py, dev.py) put
-options in THIRD_PARTY_SETTINGS. startup.py probes the ENABLE_THIRD_PARTY_AUTH.
+whether this module is enabled. startup.py probes the ENABLE_THIRD_PARTY_AUTH.
 If true, it:
 
     a) loads this module.
-    b) calls apply_settings(), passing in settings.THIRD_PARTY_AUTH.
-       THIRD_PARTY AUTH is a dict of the form
-
-       'THIRD_PARTY_AUTH': {
-            '<PROVIDER_NAME>': {
-                '<PROVIDER_SETTING_NAME>': '<PROVIDER_SETTING_VALUE>',
-                [...]
-            },
-            [...]
-       }
-
-       If you are using a dev settings file, your settings dict starts at the
-       level of <PROVIDER_NAME> and is a map of provider name string to
-       settings dict. If you are using an auth.json file, it should contain a
-       THIRD_PARTY_AUTH entry as above.
-    c) apply_settings() builds a list of <PROVIDER_NAMES>. These are the
-       enabled third party auth providers for the deployment. These are enabled
-       in provider.Registry, the canonical list of enabled providers.
-    d) then, it sets global, provider-independent settings.
-    e) then, it sets provider-specific settings. For each enabled provider, we
-       read its SETTINGS member. These are merged onto the Django settings
-       object. In most cases these are stubs and the real values are set from
-       THIRD_PARTY_AUTH. All values that are set from this dict must first be
-       initialized from SETTINGS. This allows us to validate the dict and
-       ensure that the values match expected configuration options on the
-       provider.
-    f) finally, the (key, value) pairs from the dict file are merged onto the
-       django settings object.
+    b) calls apply_settings(), passing in the Django settings
```

Comment:
I love that this PR is getting rid of the sub-settings objects, and the awkward way that they were loaded.